docs/kf_uids/uid_migration.md

```diff
@@ -0,0 +1,23 @@
+# Migrating UIDs
+
+Thoughts about UID migration.
+
+## Persist All Old Records
+
+### Translate GET requests
+
+Old style IDs should be replaced with new style IDs, but the old style IDs should become aliases pointing to the new ones.
+
+* GET requests where the requested ID looks like the old style should bounce through a translation lookup to get back the corresponding new ID and then return the entity with the new ID in place.
+* Will need a new table in database with two columns: (Old ID, New ID).
+  * When adding a new entry to that table, it should see if any existing entries have their New ID equal to the new Old ID. If so, they should be updated to have new New ID. That way all old IDs will always point to the latest ID for the corresponding entity, even if we migrate IDs multiple times.
+
+### No need to re-header existing genomic files
+
+Our harmonized genomic files have had biospecimen kfids baked into their bits (not just s3 tags). Updating those would cost extra 💰 beyond just the labor involved. (How much, I don't know. Probably not _thaaaaat_ much.)
+
+We don't need to though. As long as the GET translation is in place, the old KFIDs will continue to be valid.
```

Comment:
After some discussion about having padding bits (time = 0) and/or keeping the old kf_ids and just creating new ones, seems like the best path forward is to replace all of the kf_ids with the new format and have a mapping as described above. The main think to ensure is what the workflow looks like for the portal and for users who have genomic files in hand with BS IDs, how they would retrieve that information from the portal.

Reply:
Also just for my own notes - this doesn't apply to the study ids which will remain the old scheme.